HeadersImports.md

```diff
@@ -0,0 +1,68 @@
+# Headers and Imports
+
+## Introduction
+
+Follow this set of guidelines when creating header files and importing them. The
+guidelines are designed to support a wide range of build systems and usage scenarios.
+
+In this document, the term `library` refers to a buildable package. In CocoaPods, it's a CocoaPod.
+In Swift Package Manager, it's a library target.
+
+## Header File Types and Locations
+
+* *Public Headers* - Headers that define the library's API. They should be located in
+  `FirebaseFoo/Sources/Public`. Any additions require a minor version update. Any changes or
+  deletions require a major version update.
+
+* *Public Umbrella Header* - A single header that includes the full library's public API located at
+  `FirebaseFoo/Sources/Public/FirebaseFoo.h`. This header should be included in
+   [Firebase.h](CoreOnly/Sources/Firebase.h).
+
+* *Library Internal Headers* - Headers that are only used by the enclosing library. These headers
```

Comment:
It seems like this section should be last. It would make these ordered by public/private/internal

Reply:
Done.